classes/village/VillageManager.php

```diff
@@ -1708,6 +1726,72 @@ public static function enactProposal(System $system, User $player, int $proposal
                 self::createProposalNotification($system, $proposal['village_id'], NotificationManager::NOTIFICATION_PROPOSAL_PASSED, $proposal['name']);
                 $message = "Accepted trade offer from " . VillageManager::VILLAGE_NAMES[$proposal['target_village_id']] . "!";
                 break;
+            case self::PROPOSAL_TYPE_BEGIN_CONSTRUCTION:
```

Comment:
Will shift this entire code block and sub-functions to `ProposalManager` and use `ActionResult` instead of try/catch